lib/sitemap/sitemap_generator.rb

```diff
@@ -1,15 +1,35 @@
 class SitemapGenerator
-  EXCLUDED_FORMATS = %w[recommended-link].freeze
-
   def initialize(search_config)
     @search_config = search_config
     @all_documents = get_all_documents
   end
 
-  def self.sitemap_limit
-    25_000
+  def sitemap_chunks
+    @all_documents.each_slice(SITEMAP_LIMIT)
   end
 
+  # Generate a sitemap which matches the format specified in https://www.sitemaps.org/protocol.html
+  def generate_xml(chunk)
+    builder = Nokogiri::XML::Builder.new(encoding: "UTF-8") do |xml|
+      xml.urlset(xmlns: "http://www.sitemaps.org/schemas/sitemap/0.9") do
+        chunk.each do |document|
+          xml.url {
+            xml.loc document.url
+            xml.lastmod document.last_updated if document.last_updated
+            xml.priority document.priority
+          }
+        end
+      end
+    end
+
+    builder.to_xml
+  end
+
+private
+
+  EXCLUDED_FORMATS = %w[recommended-link].freeze
+  SITEMAP_LIMIT = 25_000
```

Comment:
We may be able to up this to 50,000 now - which is the max amount of entries one sitemap file can have.  It used to be 50,000, but when we switched to ES5 the rake task was regularly running out of memory so I halved it.

Reply:
That's interesting, is there a benefit to having fewer files?

Reply:
Not sure.  I imagine having 50,000 1-entry files would be bad, but I don't know if 1 50,000-entry file is better than 2 25,000-entry files.